### PR TITLE
bug(refs DPLAN-11472): Add event for phase selecting

### PR DESCRIPTION
**Pull request by gruenbergerdemos**
### Ticket
DPLAN-11472

- Set internal and public phase dynamically based on select
- Give data-cy as string, because component expects a string
- Add data-cy for participation phases
- Update test for participation phases
   - Mount wrapper before each test to avoid duplications in tests
   - Remove test for naming, because it is deprecated (testing implementation details is considered bad practice)
   - Add test for emitting event

### How to review/test
Verfahren -> Grundeinstellungen -> Verfahrensschritt Institiution/Verfahrensschritt Öffentlichkeit -> Based on which phase you select, "Verfahrensschritt automatisch umstellen" should be disabled or enabled (disabled for participation phases)

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [x] Tests updated/created
- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [x] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 1]
```diff
@@ -450,10 +453,10 @@
 
                                     <dp-date-range-picker
                                         class="u-2-of-4"
-                                        :data-cy="{
+                                        :data-cy="JSON.stringify({
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:456]: shouldn't this be something like `institutionPeriod` ?
The comeonent then appends `:endDate` and `startDate` 
https://github.com/demos-europe/demosplan-ui/blob/0c690b83c39d396b24c8ff2320537605e72a99b0/src/components/DpDateRangePicker/DpDateRangePicker.vue#L14

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 2]
```diff
@@ -573,10 +579,10 @@
                                     } %}
                                     <dp-date-range-picker
                                         class="u-2-of-4"
-                                        :data-cy="{
+                                        :data-cy="JSON.stringify({
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:582]: see above

File: tests/frontend/unit/ParticipationPhases.spec.js [hunk 3]
```diff
@@ -1,53 +1,47 @@
...
+          value: '1'
+        }
+      }
+    })
   })
 
-  it('should be named ParticipationPhases', () => {
-    expect(ParticipationPhases.name).toBe('ParticipationPhases')
+  it('should be instantiated', () => {
+    expect(wrapper).toBeTruthy()
```
Comment (salisdemos) [inline tests/frontend/unit/ParticipationPhases.spec.js:41]: why this change?

File: tests/frontend/unit/ParticipationPhases.spec.js [hunk 4]
```diff
@@ -1,53 +1,47 @@
...
 
-  it('sets the `inParticipatin` state correctly depending on whether the selected Phase is a participation phase', async () => {
-    const localVue = createLocalVue()
-
-    const wrapper = shallowMountWithGlobalMocks(ParticipationPhases, {
-      propsData: componentsPropsData,
-      localVue
-    })
-
+  it('should set the `inParticipatin` state correctly depending on whether the selected Phase is a participation phase', async () => {
```
Comment (salisdemos) [inline tests/frontend/unit/ParticipationPhases.spec.js:44]:
```suggestion
  it('should set the `inParticipation` state correctly depending on whether the selected Phase is a participation phase or not', async () => {
```

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 5]
```diff
@@ -64,10 +64,10 @@
             enforce-plausible-dates
             :min-date="startDate"
             required
-            :data-cy="{
+            :data-cy="JSON.stringify({
                 endDate: dataCyEndDate,
                 startDate: dataCyStartDate
-            }"
+            })"
```
Comment (sakutademos) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:70]: I think it would be better to either directly change the type to an object in the DateRangePicker or to only pass a string, such as `data-cy="newProcedureForm"`. This is because in DpDateRangePicker at [lines 14 and 28]([url](https://github.com/demos-europe/demosplan-ui/blob/main/src/components/DpDateRangePicker/DpDateRangePicker.vue#L14)), we are already passing `${dataCy}:startDate` and `${dataCy}:endDate`.

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig (hunk 1 above)
Comment (gruenbergerdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:456]: Good to know, thanks for spotting

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue (hunk 5 above)
Comment (gruenbergerdemos) [reply client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue]: Thanks, changed in [38972aa](https://github.com/demos-europe/demosplan-core/pull/3086/commits/38972aa27cd8f569f68b315b090a6107fc08b0a9)

File: tests/frontend/unit/ParticipationPhases.spec.js (hunk 3 above)
Comment (gruenbergerdemos) [reply tests/frontend/unit/ParticipationPhases.spec.js]: Good point. I now removed testing for object, too (like the test for name), because testing for implementation details is considered bad practice and provides little to no value. Name() is also deprecated in vue test utils and will be removed soon (see https://v1.test-utils.vuejs.org/upgrading-to-v1/#name). 
But I also saw that we test name and object very often. Should we discuss this with the team?

**Review by salisdemos — APPROVED** [review]
